tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/binaryoperations/TypeTestExprTest.java

```diff
@@ -47,6 +47,22 @@ public void testTypeTestExprSemanticsNegative() {
         BAssertUtil.validateError(negativeResult, 0, "unknown type 'C'", 35, 14);
     }
 
+    @Test
+    public void testTypeTestExprTypeNarrowingNegative() {
+        CompileResult negativeResult =
+                BCompileUtil.compile("test-src/expressions/binaryoperations/type-test-type-narrowing-negative.bal");
+        int i = 0;
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: expected 'SomeRecord', found 'SomeRecord?'", 49, 17);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: expected '(SomeRecord|int)', found '(SomeRecord|int)?'", 52, 21);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: expected '()', found 'SomeRecord'", 63, 13);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: expected '(SomeRecord|int)', found '(SomeRecord|int)?'", 69, 13);
```

Comment:
Is this correct? Isn't the type of `g` `SomeRecord|int` in this block? And since `w` is `SomeRecord|int`, type narrowing shouldn't be reset in the current master?